src/main/java/pl/wrapper/parking/facade/domain/ParkingServiceImpl.java

```diff
@@ -10,17 +12,37 @@
 import pl.wrapper.parking.pwrResponseHandler.PwrApiServerCaller;
 import pl.wrapper.parking.pwrResponseHandler.dto.ParkingResponse;
 
-import java.util.*;
-import java.util.function.Predicate;
-
 @Service
 @Slf4j
-record ParkingServiceImpl(PwrApiServerCaller pwrApiServerCaller, NominatimClient nominatimClient) implements ParkingService {
+record ParkingServiceImpl(PwrApiServerCaller pwrApiServerCaller, NominatimClient nominatimClient)
+        implements ParkingService {
+
+    @Override
+    public List<ParkingResponse> getAllWithFreeSpots(Boolean opened) {
+        Predicate<ParkingResponse> predicate = generatePredicateForParams(null, null, null, opened);
```

Comment:
Extract the .stream() part to another method, use it here and in getMostFreeSpots